arviz/rcparams.py

```diff
@@ -0,0 +1,266 @@
+            )
+
+    def __getitem__(self, key):
+        """Use dict getitem method."""
+        return dict.__getitem__(self, key)
+
+    def __delitem__(self, key):
+        """Raise TypeError if someone ever tries to delete a key from RcParams."""
+        raise TypeError("RcParams keys cannot be deleted")
+
+    def __repr__(self):
+        """Customize repr of RcParams objects."""
+        class_name = self.__class__.__name__
+        indent = len(class_name) + 1
+        repr_split = pprint.pformat(dict(self), indent=1, width=80 - indent).split("\n")
+        repr_indented = ("\n" + " " * indent).join(repr_split)
+        return "{}({})".format(class_name, repr_indented)
+
+    def __str__(self):
+        """Customize str/print of RcParams objects."""
+        return "\n".join(map("{0[0]:<22}: {0[1]}".format, sorted(self.items())))
+
+    def __iter__(self):
+        """Yield sorted list of keys."""
+        yield from sorted(dict.__iter__(self))
+
+    def __len__(self):
+        """Use dict len method."""
+        return dict.__len__(self)
+
+    def find_all(self, pattern):
+        """
+        Find keys that match a regex pattern.
+
+        Return the subset of this RcParams dictionary whose keys match,
+        using :func:`re.search`, the given ``pattern``.
+
+        Notes
+        -----
+            Changes to the returned dictionary are *not* propagated to
+            the parent RcParams dictionary.
+        """
+        pattern_re = re.compile(pattern)
+        return RcParams((key, value) for key, value in self.items() if pattern_re.search(key))
+
+    def copy(self):
+        """Get a copy of the RcParams object."""
+        return {k: dict.__getitem__(self, k) for k in self}
+
+
+def get_arviz_rcfile():
+    """Get arvizrc file.
+
+    The file location is determined in the following order:
+
+    - ``$PWD/arvizrc``
+    - ``$ARVIZ_DATA/arvizrc``
+    - On Linux,
+        - ``$XDG_CONFIG_HOME/arviz/arvizrc`` (if ``$XDG_CONFIG_HOME``
+          is defined)
+        - or ``$HOME/.config/arviz/arvizrc`` (if ``$XDG_CONFIG_HOME``
+          is not defined)
+    - On other platforms,
+        - ``$HOME/.arviz/arvizrc`` if ``$HOME`` is defined
+
+    Otherwise, the default defined in ``rcparams.py`` file will be used.
+    """
+    # no blank lines allowed after function docstring by pydocstyle,
+    # but black requires white line before function
+
+    def gen_candidates():
+        yield os.path.join(os.getcwd(), "arvizrc")
+        arviz_data_dir = os.environ.get("ARVIZ_DATA")
+        if arviz_data_dir:
+            yield os.path.join(arviz_data_dir, "arvizrc")
+        xdg_base = os.environ.get("XDG_CONFIG_HOME", str(Path.home() / ".config"))
+        if sys.platform.startswith(("linux", "freebsd")):
+            configdir = str(Path(xdg_base, "arviz"))
+        else:
+            configdir = str(Path.home() / ".arviz")
+        yield os.path.join(configdir, "arvizrc")
+
+    for fname in gen_candidates():
```

Comment:
i'm a `pathlib` convert since yesterday, but it might be nice for `gen_candidates` to yield `Path` objects so you don't have to cast back and forth, and you can just call `if fname.exists() and not fname.isdir():  return str(fname)`